src/software/thunderscope/replay/replay_file_debugging_script.py

```diff
@@ -0,0 +1,71 @@
+import base64
+import glob
+import gzip
+import os
+import argparse
+
+from proto.import_all_protos import *
+from software.thunderscope.replay.replay_constants import *
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Script for debugging replay files"
+    )
+    parser.add_argument(
+        "--replay_dir",
+        action="store",
+        help=f"Folder containing the .{REPLAY_FILE_EXTENSION} files",
+        required=True,
+        type=os.path.abspath,
+    )
+    args = parser.parse_args()
+
+    # Load up all replay files in the log folder
+    replay_files = glob.glob(args.replay_dir + f"/*.{REPLAY_FILE_EXTENSION}")
+
+    # Sort the files by their chunk index
+    def __sort_replay_chunks(file_path: str):
+        head, tail = os.path.split(file_path)
+        replay_index, _ = tail.split(".")
+        return int(replay_index)
+
+
+    sorted_chunks = sorted(replay_files, key=__sort_replay_chunks)
+
+    print(f"Found {len(sorted_chunks)} replay files in {args.replay_dir}")
+
+    total_line_num = 0
+    for replay_file_name in sorted_chunks:
+        print(f"Replaying {replay_file_name}...")
+        with gzip.open(replay_file_name, "rb") as replay_file:
+            line = replay_file.readline()
+            line_num = 0
+            while line is not None:
```

Comment:
we should ignore blank lines: '' because this code breaks often if there are blank lines in the proto log replay (which is quite often)

Reply:
Unless the replay file is corrupted, there shouldnt be blank lines. All lines should have the `timestamp, protobuf_type, data` format, and if they don't, then it's probably good to know when debugging using this script.

Reply:
If I take any ai vs ai replay and try to replay it with the script, it always fails on 0.replay even when proto player seems fine. It fails because the replay has a blank line in it.

Reply:
Okay I just tried it and got the same error. That's weird. @Mr-Anyone could you add a check to see if `line == b""` and skip over those (and maybe just log something) so we avoid getting the `ValueError` when we `.split`?